ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/strategy/transferstrategy/AbstractTransferStrategy.java

```diff
@@ -167,24 +186,45 @@ private synchronized void maybeSetCachedException(final Throwable t) {
 
     private void transferJobParts(final Iterable<JobPart> jobParts,
                                   final CountDownLatch countDownLatch,
-                                  final AtomicInteger numBlobsTransferred) throws IOException {
+                                  final AtomicInteger numBlobsTransferred,
+                                  final BooleanSupplier transferPredicate)
+    {
+        final AtomicInteger numJobPartsToTransfer = new AtomicInteger(Iterables.size(jobParts));
+
         for (final JobPart jobPart : jobParts) {
-            executorService.execute(() -> {
-                try {
+            if (executorService.isShutdown()) {
+                LOG.debug("Executor service is shut down, decrementing countdown latch");
+                countDownLatch.countDown();
+            } else {
+                maybeSubmitJobPartTransfer(countDownLatch, numBlobsTransferred, transferPredicate, numJobPartsToTransfer, jobPart);
+            }
+        }
+    }
+
+    private void maybeSubmitJobPartTransfer(final CountDownLatch countDownLatch, final AtomicInteger numBlobsTransferred, final BooleanSupplier transferPredicate, final AtomicInteger numJobPartsToTransfer, final JobPart jobPart) {
+        executorService.submit((Callable<Void>) () -> {
```

Comment:
Do we currently support cancelling tasks that are mid process?  It doesn't look like we are given that we don't capture the future and call cancel on it.